lib/web_ui/dev/test_runner.dart

```diff
@@ -448,69 +513,69 @@ class TestCommand extends Command<bool> with ArgUtils {
     timestampFile.writeAsStringSync(timestamp);
   }
 
+  Future<void> _buildTestsInParallel(
+      {List<FilePath> targets, bool forCanvasKit = false}) async {
+    final List<TestBuildInput> buildInputs = targets
+        .map((FilePath f) => TestBuildInput(f, forCanvasKit: forCanvasKit))
+        .toList();
+
+    final results = _pool.forEach(
+      buildInputs,
+      _buildTest,
+    );
+    await for (final bool isSuccess in results) {
+      if (!isSuccess) {
+        throw ToolException('Failed to compile tests.');
+      }
+    }
+  }
+
   /// Builds the specific test [targets].
   ///
   /// [targets] must not be null.
   ///
-  /// When building for CanvasKit we have to use a separate `build.canvaskit.yaml`
-  /// config file. Otherwise, `build.html.yaml` is used. Because `build_runner`
-  /// overwrites the output directories, we redirect the CanvasKit output to a
-  /// separate directory, then copy the files back to `build/test`.
-  Future<void> _buildTests({List<FilePath> targets, bool forCanvasKit}) async {
-    print(
-        'Building ${targets.length} targets for ${forCanvasKit ? 'CanvasKit' : 'HTML'}');
-    final String canvasKitOutputRelativePath =
-        path.join('.dart_tool', 'canvaskit_tests');
+  /// Uses `dart2js` for building the tests.
+  ///
+  /// When building for CanvasKit we have to use extra argument
+  /// `DFLUTTER_WEB_USE_SKIA=true`.
+  Future<bool> _buildTest(TestBuildInput input) async {
+    final targetFileName = input.target.relativeToWebUi
+        .replaceFirst('.dart', '.dart.browser_test.dart.js');
```

Comment:
Why `replaceFirst`? Isn't `.dart` the last part of the file name? If so, then should be the same as appending ".browser_test.dart.js" at the end, i.e. `'${input.target.relativeToWebUi}.browser_test.dart.js'`

Reply:
Yes it is the same. You are right, your suggestion is probably even more performant (no patterns to search, for example).

Reply:
done